distributed/dashboard/scheduler.py

```diff
@@ -268,8 +272,19 @@ def collect(self):
             "Number of tasks known by scheduler.",
             labels=["state"],
         )
-        tasks.add_metric(["received"], len(self.server.tasks))
-        tasks.add_metric(["unrunnable"], len(self.server.unrunnable))
+
+        task_counter = merge_with(
+            sum, (tp.states for tp in self.server.task_prefixes.values())
+        )
+
+        yield CounterMetricFamily(
+            "dask_scheduler_tasks_forgotten",
+            "Number of tasks already forgotten by scheduler.",
```

Comment:
I wonder if something like"total processed tasks would be more informative?"

We can still pop off the forgotten value at the end, but I suspect that people would be interested in the total of memory + forgotten + erred or something similar, more than just forgotten.

Reply:
> I wonder if something like"total processed tasks would be more informative?"

As far as I understand, the approach with prometheus should rather be to export raw metric data. Interpretation, aggregation, etc. can then be done by the user. With these metrics everything is exposed and can be combined as needed. The way I see it, the bokeh dashboards are there to provide "easily understandable" information with context but for prometheus I would rather go for "raw data".

The reasoning for the metrics here is that the gauge `dask_scheduler_tasks` describes what the scheduler currently actually cares about and what it is working with. A more appropriate name would maybe be `dask_scheduler_active_tasks` but I'm also fine to stick to the current name.
By removing the `forgotten` state from this gauge there is also a nice invariant `sum(dask_scheduler_tasks) == len(scheduler.tasks)` and the possibility to infer progress with these states.

I split off the `forgotten` state since it is technically not a gauge in the prometheus sense (increasing *and* decreasing) but rather a count (only increasing). Ultimately it's a measure of how much work the scheduler has done since it was started.

Reply:
> As far as I understand, the approach with prometheus should rather be to export raw metric data

Fair enough.  Is there some better term than forgotten?  This seems a bit negative, as though something has gone wrong.  I can imagine people raising issues "why did Dask forget so many of my tasks?" in the future.

Reply:
I'm also happy to merge this as-is though.  My comments here are mostly academic.

Reply:
> Fair enough. Is there some better term than forgotten?

The term `done` comes to my mind since (if everything behaves as expected) a task is transitioned to `forgotten` once it is no longer needed, i.e. `done`. More descriptive would be `successfully processed` similar to what you already suggested.
I wanted to stick to the internal state name to avoid confusion if somebody actually bothers to look into the code :)
I will change the documentation to better describe what this actually means.

Reply:
I was initially also struggling with all of the states and am wondering if they are documented anywhere. Since these states are also visualised on the dashboard it probably makes sense to include it in the official docs, e.g. a section in https://distributed.dask.org/en/latest/journey.html

Reply:
There's a description of the different task states at https://distributed.dask.org/en/latest/scheduling-state.html#task-state